src/core/matter/mod.rs

```diff
@@ -1,22 +1,22 @@
 use base64::{engine::general_purpose as b64_engine, Engine};
-#[cfg(feature = "python")]
-use pyo3::prelude::pyclass;
 
 use crate::core::util;
 use crate::error::{err, Error, Result};
 
 pub mod tables;
 
-#[cfg_attr(feature = "python", pyclass)]
-#[derive(Debug, Clone)]
-pub struct Matter {
-    pub(crate) raw: Vec<u8>,
-    pub(crate) code: String,
-    pub(crate) size: u32,
-}
-
-impl Matter {
-    pub fn new_with_code_and_raw(code: &str, raw: &[u8]) -> Result<Matter> {
+pub trait Matter: Default {
+    fn code(&self) -> String;
+    fn size(&self) -> u32;
+    fn raw(&self) -> Vec<u8>;
+    fn set_code(&mut self, code: &str);
```

Comment:
I suppose these `set_` methods must be crate private but now they will be public and any Matter instance can be modified.
You rather need to add a basic constructor method and use it instead of setters.

Reply:
I cam make that change in the PR I'm working on now, #93. It make sense.

Reply:
I actually wondered if `parside` would need some things here.

Reply:
Anyhow I think I made the change you suggested, basically to change all the traits to `pub(crate)`.

Reply:
> Anyhow I think I made the change you suggested, basically to change all the traits to `pub(crate)`.

This is a bad change.  You will not be able to call any methods defined in `Matter` trait out of the `cesride`. 
For instance: in order to call `Dater::new_with_qb64` in another library you need to import `Matter` trait but you made it private.

Reply:
Yeah, that's why I said above:
> I actually wondered if parside would need some things here.

So it's obvious I do not understand what you meant, but instead of clarifying you simply told me why what I did _wouldn't_ work. In the future, dropping a piece of code or clarifying comment to explain what you mean would be helpful.

That change I made isn't necessarily bad, but it requires more code to wrap all the matter methods in the concrete classes which I haven't done (if we want to expose things outside `cesride`).

Reply:
I am a new Rust coder, but I am very familiar with C and many other languages, if that helps you communicate.

Reply:
I apologize if you found my comment rude. I'll take it into account.

Reply:
I think I had misinterpreted what you meant to be the opposite, also saw the PR, that is how I understood.

Reply:
Not as much rude but if you know the solution or why I am misunderstanding something, helping is the open source thing to do!